_studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp

```diff
@@ -687,6 +687,23 @@ mfxStatus VideoDECODEH265::DecodeHeader(VideoCORE *core, mfxBitstream *bs, mfxVi
     umcRes = FillParam(core, &decoder, par, false);
     MFX_CHECK(umcRes == UMC::UMC_OK, ConvertUMCStatusToMfx(umcRes));
 
+    mfxExtCodingOptionVPS * extVps = (mfxExtCodingOptionVPS *)GetExtendedBuffer(par->ExtParam, par->NumExtParam, MFX_EXTBUFF_CODING_OPTION_VPS);
+    if (extVps)
+    {
+        RawHeader_H265 *vps = decoder.GetVPS();
+
+        if (vps->GetSize())
+        {
+            MFX_CHECK(extVps->VPSBufSize >= vps->GetSize(), MFX_ERR_NOT_ENOUGH_BUFFER);
+            extVps->VPSBufSize = (mfxU16)vps->GetSize();
+            std::copy(vps->GetPointer(), vps->GetPointer() + extVps->VPSBufSize, extVps->VPSBuffer);
+        }
```

Comment:
@dgurulev , can you please review this part? Should we save info if vps was changed and use this info here https://github.com/Intel-Media-SDK/MediaSDK/blob/dba2964cf1953404ee5ffb17cc3fb1b079da91a5/_studio/shared/umc/codec/h265_dec/src/umc_h265_task_supplier.cpp#L2224 (as already done for SPS/PPS to close AUs)?

Reply:
@dmitryermilov,

I'm not sure that VPS is permitted somewhere excepting the beginning of a bitstream, need to refer to spec.